sensor/docs/tutorials/tag_based_pnp_calibrator.md

```diff
@@ -0,0 +1,153 @@
+
+![tag_based_pnp_calibrator](../images/tag_based_pnp_calibrator/visualization3.jpg)
+
+The output in the console should be as follows:
+
+```text
+[tag_based_pnp_calibrator_node]:  Initial reprojection error=15.41
+[tag_based_pnp_calibrator_node]:  Current reprojection error=1.09
+[tag_based_pnp_calibrator_node]:  Filtered reprojection error=1.10
+```
+
+## Results
+
+After the calibration process finishes, the `sensor_calibration_manager` will display the results in the UI and allow the user to save the calibration data to a file.
+
+In the UI of the X2 project, three different TF trees are displayed: `Initial TF Tree`, `Calibration Tree`, and `Final TF Tree`. The `Initial TF Tree` presents the initial TF connections between sensors needed for calibration. The `Calibration Tree` shows the calibrated transformation between sensors, in this tutorial, `camera6/camera_optical_link` and `pandar_40p_front`. The `Final TF Tree` depicts the TF tree after incorporating the updated calibrated transformation. Since the transformations represented by the black arrows are fixed, the transformation between `front_unit_base_link` and `camera6/camera_link`, which is represented by the red arrow, can be calculated using the calibrated transformation.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/menu4.jpg" alt="menu4" width="500">
+</p>
+
+During the calibration or at the end of the calibration, the user can modify the `visualization options` on the right side of the `image view`. Setting the `Marker size (m)` to `0.04` and setting the `PC subsample factor` to `1` can help the user visualize the projected pointcloud on the image.
+
+<p align="center">
+    <img src="../images/tag_based_pnp_calibrator/visualization_bar.jpg"  alt="visualization_bar" width="200">
+</p>
+
+After setting the options above, change the `/initial_tf` (in the `visualization options`) to `/current_tf` to measure the difference after the calibration.
+
+The images below show that with the calibrated transformation, the projected pointcloud aligns with the image.
+
+<table>
+  <tr>
+    <td><img src="../images/tag_based_pnp_calibrator/init_tf.jpg" alt="init_tf" width = 500px ></td>
+    <td><img src="../images/tag_based_pnp_calibrator/calibrated_tf.jpg" alt="calibrated_tf" width = 500px ></td>
+   </tr>
+   <tr>
+    <td><p style="text-align: center;">Before Calibration.</p></td>
+    <td><p style="text-align: center;">After Calibration.</p></td>
+  </tr>
+</table>
+
+## FAQ
+
+- Why doesn't the calibrator add calibration pairs?
+
+  1. One possible reason is that the current pair is too close to previously collected data. In that case, the current data is not accepted.
+  2. The timestamps of the lidar and camera are not synchronized, this can be checked with `ros2 topic echo [topic_name] --field header.stamp`. Setting the parameter `use_receive_time` to `True` might help to solve the issue.
```

Comment:
```suggestion
  2. The timestamps of the lidar and camera are not synchronized, this can be checked with `ros2 topic echo [topic_name] --field header.stamp`. Setting the parameter `use_receive_time` to `True` might help to solve the issue but is not recommended as a long term solution.
```